src/vsr/replica.zig

```diff
@@ -302,6 +302,12 @@ pub fn ReplicaType(
         /// Invariants:
         /// - If syncing≠idle then sync_tables=null.
         sync_tables: ?ForestTableIterator = null,
+        /// Tracks wal repair progress to decide when to switch to state sync.
+        /// Updated on repair_sync_timeout.
+        sync_wal_repair_progress: struct {
+            commit_min: u64 = 0,
+            unchanged: bool = false,
+        } = .{},
```

Comment:
Could we get by with just storing `commit_min`, and computing `unchanged` via `self.sync_wal_repair_progress.commit_min == self.commit_min`?

Reply:
I think that would be prone to false positives. It could be the case that you invoke `repair_stuck` just after the timeout triggered and updated `commit_min`. 

But I think I see the bug in the opposite direction. If at the point of `repair_stuck` we see `.unchanged`, but our commit_min advanced, than we are not, in fact, stuck. So we kinda need both  I think!

Reply:
Ah, but let me remove negation here, `!unchanged` is unnecessary baroque!